### PR TITLE
issue 858 added code to validate ServerSideConfiguration.  Also included unit tests

### PR DESCRIPTION
**Pull request by gibsong**
https://github.com/ehcache/ehcache3/issues/858

### CODE REVIEW

File: clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java [hunk 1]
```diff
@@ -393,16 +393,79 @@ private EhcacheEntityResponse validate(ClientDescriptor clientDescriptor, Valida
       return failure(new IllegalStateException("Clustered Store Manager is not configured"));
     }
     ServerSideConfiguration serverSideConfiguration = message.getConfiguration();
-    if (!this.sharedResourcePools.keySet().equals(serverSideConfiguration.getResourcePools().keySet())) {
-      return failure(new IllegalArgumentException("ResourcePools not aligned"));
-    } else if (!nullSafeEquals(this.defaultServerResource, serverSideConfiguration.getDefaultServerResource())) {
+
+    if(isInheritedConfig(serverSideConfiguration))
+    {
```
Comment (ljacomet) [inline clustered/server/src/main/java/org/ehcache/clustered/server/EhcacheActiveEntity.java]: Please, use code formatting existing in the project

**Comment by ljacomet** [comment]
Test failures -  needs fixing

**Comment by ljacomet** [comment]
Also reword that commit message, in its current form, it is useless

**Comment by gibsong** [comment]
test this please

**Comment by ljacomet** [comment]
Merge on the command line after a rebase

**Pull request closed without merging**